apollo-composition/src/lib.rs

```diff
@@ -133,23 +134,27 @@ pub trait HybridComposition {
                 self.update_supergraph_sdl(raw_sdl);
                 let satisfiability_result = self.validate_satisfiability().await;
                 self.add_issues(
-                    satisfiability_result_into_issues(satisfiability_result)
-                        .map(|mut issue| {
-                            for (service_name, connector) in by_service_name.iter() {
-                                issue.message = issue.message.replace(
-                                    &**service_name,
-                                    connector.id.subgraph_name.as_str(),
-                                );
-                            }
-                            issue
-                        })
-                        .chain(once(Issue {
+                    satisfiability_result_into_issues(satisfiability_result).map(|mut issue| {
+                        for (service_name, connector) in by_service_name.iter() {
+                            issue.message = issue
+                                .message
+                                .replace(&**service_name, connector.id.subgraph_name.as_str());
+                        }
+                        issue
+                    }),
+                );
+
+                if env::var("CONNECTORS_PREVIEW").unwrap_or_default() != "true" {
```

Comment:
Small nit, but do we want users to be able to use this env var like others? If so, we need to handle cases where it is equal to other truthy things (e.g. `1`, `"1"`, `TRUE`, etc.). The router gets this functionality by using clap's [`FalseyValueParser`](https://docs.rs/clap/latest/clap/builder/struct.FalseyValueParser.html).

Reply:
i'm really just trying to avoid some misleading stuff in demos. i actually can't figure out how to set this env var in the appropriate place. i might just make a custom build for demos that removes this

Reply:
Super fair. Feel free to wrap this in maybe a `#[cfg(debug_assertions)]` or a custom feature if you want!

Reply:
I think if this is controlled via env var, we probably want to set this on the supergraph binary itself. Then we can use the same Clap options as we do for other env vars

Reply:
That would mean making it a Rust option here—which could also be set in the web editor if we want.

Do we just want to get rid of or change this warning, though?

Reply:
yeah maybe. warning logs in router might be better now that composition errors show up in more tools